sled-agent/src/services.rs

```diff
@@ -5039,6 +5101,18 @@ mod illumos_tests {
             .await
             .expect("Failed to ensure disks");
         assert!(!result.has_error(), "{:?}", result);
+        let result = harness
```

Comment:
This is necessary now because we want to read the dataset ledger for our service tests, even if it's empty.